pandas/core/frame.py

```diff
@@ -2212,6 +2212,63 @@ def insert(self, loc, column, value, allow_duplicates=False):
         self._data.insert(
             loc, column, value, allow_duplicates=allow_duplicates)
 
+    def assign(self, **kwargs):
+        """
+        Assign new columns to a DataFrame.
+
+        Parameters
+        ==========
```

Comment:
can you use a single line `----` here? (for consistency with our other docstrings)

also we generally use no blankline after a heading

(see https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt for an overview)